File: LeetCode/remote/excel_sheet_column_number.rb
#   https://leetcode.com/problems/excel-sheet-column-number/
#   https://leetcode.com/submissions/detail/54024927/

def title_to_number(s)
    s.chars.inject(0) { |acc, x| acc * 26 + (x.ord - 'A'.ord + 1) }
end

File: LeetCode/remote/excel_sheet_column_number.cs
//  https://leetcode.com/problems/excel-sheet-column-number/
//  I fucking hate this problem
//  https://leetcode.com/submissions/detail/60819887/
//
//  Submission Details
//  1000 / 1000 test cases passed.
//      Status: Accepted
//      Runtime: 136 ms
//          
//          Submitted: 0 minutes ago
public class Solution {
    public int TitleToNumber(string s) {
        return s.Aggregate(0, (acc, x) => acc * 26 + x - 'A' + 1);
    }
}
